### ISSUE TITLE
`flake8-bugbear.extend-immutable-calls` does not work with relative imports in sub-packages

### ISSUE THREAD

**Issue by JuneStepp** [astral-sh/ruff#18358]
### Summary

I've attached a minimal reproducible example to this issue, including the ruff.toml file. The `does_not_work.py` file shows that `extend-immutable-calls` does not prevent a linting error when the function is imported using a relative import from a sub-package. The two `works.py` files demonstrate that it does work with both relative imports within the main package and with absolute imports in a sub-package The command run is `ruff check`.

[example.zip](https://github.com/user-attachments/files/20495754/example.zip)

### Version

ruff 0.11.11

**Comment by ntBre** [comment]
This sounds plausible to me because I think `extend-mutable-calls` does a pretty simple comparison that may not match the relative path correctly. However, I'm having some trouble with the attached example. I unzipped the file and ran

```shell
ruff check
```

which only reports two `RUF009` errors since that's the only rule selected in `ruff.toml`. When I select the `flake8-bugbear` rules with

```shell
ruff check --select B
```

I don't get any diagnostics at all.

Could you provide a reproducing example? I'd also slightly prefer a couple of markdown code blocks, if possible so I can avoid downloading a file 😅

**Comment by JuneStepp** [comment]
> which only reports two `RUF009` errors since that's the only rule selected in `ruff.toml`. 

That's the correct error. The example uses dataclasses. `RUF009` is also affected by `extend-immutable-calls` as documented at https://docs.astral.sh/ruff/settings/#lint_flake8-bugbear_extend-immutable-calls.

**Comment by ntBre** [comment]
Oh I see, I missed that in the docs, thanks! I'll take a closer look at the reproduction then.

**Comment by ntBre** [comment]
Just to make this a little easier to glance at in the future, this is the project structure in the `example` directory:

```
.
├── package
│   ├── custom_field.py
│   ├── sub_package
│   │   ├── does_not_work.py
│   │   └── works.py
│   └── works.py
└── ruff.toml
```

And these are the contents of the files, with comments added showing where RUF009 is emitted.

### `ruff.toml`

```toml
[lint]
select = ["RUF"]
flake8-bugbear.extend-immutable-calls = ["package.custom_field.custom_field"]
```

### `package/custom_field.py`

```python
def custom_field() -> list[int]:
    return [1, 2, 3, 4]
```

### `package/sub_package/does_not_work.py`

```python
from dataclasses import dataclass

from ..custom_field import custom_field


@dataclass
class A:
    some_field: list[int] = custom_field()  # RUF009
```

### `package/sub_package/works.py`

```python
from dataclasses import dataclass

from package.custom_field import custom_field


@dataclass
class A:
    some_field: list[int] = custom_field()
```

### `package/works.py`

```python
from dataclasses import dataclass

from .custom_field import custom_field


@dataclass
class A:
    some_field: list[int] = custom_field()  # RUF009
```

If I add an empty `package/__init__.py` file, I get the error in `package/works.py` to go away. Could that be what you were seeing before?

In that case, the false positive would be with the `from ..custom_field import custom_field` import in `package/sub_package/does_not_work.py`.

As a minor note, everything still seems to reproduce even if I delete `package/custom_field.py`, which might be a slight help to someone turning this into a CLI test in the future.

**Comment by JuneStepp** [comment]
> If I add an empty package/__init__.py file, I get the error in package/works.py to go away. Could that be what you were seeing before?

Probably.
